modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/config/OperationBeanDefinitionParser.java

```diff
@@ -0,0 +1,191 @@
+
+                @Override
+                public void onOperation()
+                {
+                    nestedOperations.put(parameter.getName(), parseNestedProcessor(element, parameter, parserContext));
+                }
+
+                @Override
+                public void onList()
+                {
+                    DataType[] generics = type.getGenericTypes();
+                    if (!ArrayUtils.isEmpty(generics) && generics[0].getQualifier() == DataQualifier.OPERATION)
+                    {
+                        nestedOperations.put(parameter.getName(), parseNestedProcessor(element, parameter, parserContext));
+                    }
+                }
+            });
+        }
+
+        return nestedOperations;
+    }
+
+    private void parseConfigRef(Element element, BeanDefinitionBuilder builder)
+    {
+        String configRef = element.getAttribute(ATTRIBUTE_NAME_CONFIG);
+        if (StringUtils.isBlank(configRef))
+        {
+            throw new IllegalArgumentException("All operations must provide a config-ref element");
+        }
+
+        builder.addConstructorArgValue(new RuntimeBeanReference(configRef));
+    }
+
+
+    private String generateChildBeanName(Element element)
+    {
+        String id = SpringXMLUtils.getNameOrId(element);
+        if (StringUtils.isBlank(id))
+        {
+            String parentId = SpringXMLUtils.getNameOrId(((Element) element.getParentNode()));
+            return String.format(".%s:%s", parentId, element.getLocalName());
+        }
+        else
+        {
+            return id;
+        }
+    }
+
+    private ManagedList<MessageProcessor> parseNestedProcessor(Element element, Parameter parameter, ParserContext parserContext)
+    {
+        element = DomUtils.getChildElementByTagName(element, NameUtils.hyphenize(parameter.getName()));
+        BeanDefinitionBuilder builder = BeanDefinitionBuilder.rootBeanDefinition(MessageProcessorChainFactoryBean.class);
+        BeanDefinition beanDefinition = builder.getBeanDefinition();
+        String childBeanName = generateChildBeanName(element);
+        parserContext.getRegistry().registerBeanDefinition(childBeanName, beanDefinition);
+        element.setAttribute("name", childBeanName);
+
+        builder.getRawBeanDefinition().setSource(parserContext.extractSource(element));
+        builder.setScope(BeanDefinition.SCOPE_SINGLETON);
+
+        ManagedList<MessageProcessor> processors = (ManagedList<MessageProcessor>) parserContext.getDelegate().parseListElement(element, builder.getBeanDefinition());
+        parserContext.getRegistry().removeBeanDefinition(generateChildBeanName(element));
+
+        return processors;
+    }
+
+    private void attachProcessorDefinition(ParserContext parserContext, BeanDefinition definition)
+    {
+        MutablePropertyValues propertyValues = parserContext.getContainingBeanDefinition()
+                .getPropertyValues();
+        if (parserContext.getContainingBeanDefinition()
```

Comment:
Yes... this method I borrowed from Devkit when seeing how were they generating their parsers, so I'm not that worried since this has been working since forever.

I do however 100% agree with you and created an issue to track this down: MULE-8732